lib/utils/class-to-invokable.ts

```diff
@@ -1,14 +1,14 @@
-'use strict';
+type Proxify = typeof Proxy;
 
 /**
  * Wraps a constructor to not need the `new` keyword using a proxy.
  * Only used for data types.
  *
- * @param {Function} Class The class instance to wrap as invocable.
+ * @param {ProxyConstructor} Class The class instance to wrap as invocable.
  * @returns {Proxy} Wrapped class instance.
  * @private
  */
-function classToInvokable(Class) {
+export function classToInvokable(Class: any): Proxify {
```

Comment:
This is a bit too hardcore for my understanding. Theoretically, `Class` should be of type `ProxyConstructor` but that causes problems with the spread operator: 

```A spread argument must either have a tuple type or be passed to a rest parameter.```

Reply:
Something like this should work:

```ts
export function classToInvokable<
  T extends (new () => any) // the () might not be necessary, can't remember. 
>(Class: T): T extends new (...infer Args) => infer Instance 
  ? (...args: Args) => Instance & (new (...args: Args) => Instance) & (Args['length'] extends 0 ? Instance : {})
   : never // this won't ever if  T is a constructor, its just here since we need an `extends` to use `infer`.
 { /* function body */}
```

(haven't verified, may need a bit of tweaking).

Reply:
what.. the... hell 😂